lib/board.rb

```diff
@@ -0,0 +1,65 @@
+class Board
+  attr_accessor :cells, :turn_count, :exes, :oes
+
+  def initialize
+    @cells = Array.new(9, " ")
+    @exes = []
+    @oes = []
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(input)
+    index = input.to_i - 1
+    @cells[index]
+  end
+
+  def exes
+    @exes = cells.map.with_index.map { |mark, i| i if mark == "X" }
+  end
+
+  def oes
+    @oes = cells.map.with_index.map { |mark, i| i if mark == "O" }
+  end
+
+  def full?
+    return false if @cells.any? { |e| e == " " }
```

Comment:
Use the most expressive and efficient iterator for a problem. `Board#full?` is asking if all elements are not empty strings. You can use logic that is symmetrical to those semantics with the [`#all?`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) iterator.

`cells.all? {|cell| cell != " "}`

This is also more efficient in that the loop will stop after the iteration for which a cell is not empty.